src/atomvm_dialyzer_provider.erl

```diff
@@ -296,22 +284,37 @@ default_base_beam_path() ->
 
 %% @private
 get_base_beam_path_list(Base) ->
+    NotFound = libs_not_found(),
     Libs =
         case [filelib:wildcard(Base ++ "/lib/" ++ atom_to_list(Lib) ++ "*/ebin") || Lib <- ?LIBS] of
-            [] ->
-                [filename:join(Base, atom_to_list(Lib2) ++ "/src/beams") || Lib2 <- ?LIBS];
+            NotFound ->
```

Comment:
This logic could be simplified or improved.
Do we want to add for each lib in ?LIBS either ebin or src/beams?

I would suggest doing a foldl on the list instead than matching with [[], [], ..., []].

Reply:
Indeed. I believe this is much cleaner now.